packages/socket-mode/src/SocketModeClient.ts

```diff
@@ -402,27 +112,263 @@ export class SocketModeClient extends EventEmitter {
   }
 
   /**
-   * End a Socket Mode session. After this method is called no messages will be sent or received unless you call
-   * start() again later.
+   * End a Socket Mode session. After this method is called no messages will be sent or received
+   * unless you call start() again later.
    */
   public disconnect(): Promise<void> {
     return new Promise((resolve, reject) => {
-      this.logger.debug('manual disconnect');
-
-      // resolve (or reject) on disconnect
+      this.logger.debug('Manually disconnecting this Socket Mode client');
+      // Resolve (or reject) on disconnect
       this.once('disconnected', (err) => {
         if (err instanceof Error) {
           reject(err);
         } else {
           resolve();
         }
       });
-
-      // delegate behavior to state machine
+      // Delegate behavior to state machine
       this.stateMachine.handle('explicit disconnect');
     });
   }
 
+  // --------------------------------------------
+  // Private methods / properties
+  // --------------------------------------------
+
+  /**
+   * State machine that backs the transition and action behavior
+   */
+  private stateMachine: StateMachine<string, string>;
+
+  /* eslint-disable @typescript-eslint/indent, newline-per-chained-call */
+
+  private connectingSubmachine: Configuration<string, string> = Finity.configure()
```

Comment:
connectingSubmachine and refreshingSubmachine are mostly the same. Perhaps, we can have a factory method, which returns a new Configuration instance instead of duplicated code. Thoughts? @stevengill